packages/sap-adapter/src/config.ts

```diff
@@ -0,0 +1,130 @@
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import _ from 'lodash'
+import { ElemID, CORE_ANNOTATIONS } from '@salto-io/adapter-api'
+import { createMatchingObjectType } from '@salto-io/adapter-utils'
+import { client as clientUtils, config as configUtils, elements } from '@salto-io/adapter-components'
+import { SAP } from './constants'
+
+const { createClientConfigType } = clientUtils
+const {
+  createUserFetchConfigType,
+  createSwaggerAdapterApiConfigType,
+} = configUtils
+
+export const CLIENT_CONFIG = 'client'
+export const FETCH_CONFIG = 'fetch'
+export const API_DEFINITIONS_CONFIG = 'apiDefinitions'
+
+export type SAPClientConfig = clientUtils.ClientBaseConfig<clientUtils.ClientRateLimitConfig>
+
+export type SAPFetchConfig = configUtils.UserFetchConfig
+
+export type SAPApiConfig = configUtils.AdapterSwaggerApiConfig
+
+export type SAPConfig = {
+  [CLIENT_CONFIG]?: SAPClientConfig
+  [FETCH_CONFIG]: SAPFetchConfig
+  [API_DEFINITIONS_CONFIG]: SAPApiConfig
+}
+
+const DEFAULT_ID_FIELDS = ['name']
+export const FIELDS_TO_OMIT: configUtils.FieldToOmitType[] = []
+
+const DEFAULT_TYPE_CUSTOMIZATIONS: SAPApiConfig['types'] = {
+  MCMService_EnergySourceTypes: {
+    request: {
+      url: 'EnergySourceTypes',
+    },
+  },
+  MCMService_MeasuringTypes: {
+    request: {
+      url: 'MeasuringTypes',
+    },
+  },
+  MCMService_MCIRateTypes: {
+    request: {
+      url: 'MCIRateTypes',
+    },
+  },
+  MCMService_PowerRangeTypes: {
+    request: {
+      url: 'PowerRangeTypes',
+    },
+  },
+  MCMService_MCMFormulas: {
+    request: {
+      url: 'MCMFormulas',
+    },
+  },
+}
+
+const DEFAULT_SWAGGER_CONFIG: SAPApiConfig['swagger'] = {
+  url: 'https://raw.githubusercontent.com/salto-io/adapter-swaggers/main/sap/swagger.json',
+}
+
+export const SUPPORTED_TYPES = {
+  MCMService_EnergySourceTypes: ['EnergySourceTypes'],
+  MCMService_MeasuringTypes: ['MeasuringTypes'],
+  MCMService_MCIRateTypes: ['MCIRateTypes'],
+  MCMService_PowerRangeTypes: ['PowerRangeTypes'],
+  MCMService_MCMFormulas: ['MCMFormulas'],
+}
+
+export const DEFAULT_API_DEFINITIONS: SAPApiConfig = {
+  swagger: DEFAULT_SWAGGER_CONFIG,
+  typeDefaults: {
+    transformation: {
+      idFields: DEFAULT_ID_FIELDS,
+      fieldsToOmit: FIELDS_TO_OMIT,
+    },
+  },
+  types: DEFAULT_TYPE_CUSTOMIZATIONS,
+  supportedTypes: SUPPORTED_TYPES,
+}
+
+export const DEFAULT_CONFIG: SAPConfig = {
+  [FETCH_CONFIG]: {
+    ...elements.query.INCLUDE_ALL_CONFIG,
+  },
```

Comment:
may be worth adding
```suggestion
    hideTypes: true,
  },
```
(unless you prefer to keep the types visible?)